methods/enumerables.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index], index
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    answer = []
+    my_each do |item|
+      answer.push(item) if yield item
+    end
+    answer
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+
+  def my_any?
+    return true unless block_given?
+
+    my_each do |item|
+      return true unless yield item
+    end
+    false
+  end
+
+  def my_none?
+    return true unless block_given?
+
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
```

Comment:
## my_none?:

- support ranges for my_none?.

- when no block or argument is given returns false only if one of the collection members is true:

```ruby

     true_array = [1, 'hi', true]
     true_array.my_none? === false => false

```

- when a class is passed as an argument returns true if none of the collection is a member of such class.

- when a class is passed as an argument returns false if any of the collection is a member of such class.

- when a Regex is passed as an argument returns true if none of the collection matches the Regex.

- when a Regex is passed as an argument returns false if any of the collection matches the Regex.

- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.

- when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern.